### ISSUE TITLE
Unavailable services when deploying on kubernetes

### ISSUE THREAD

**Issue by FarhanSajid1** [go-micro/go-micro#2458]
Hello, I am seeing an issue whenever I deploy to k8s, the connections appear to timeout and I believe this is due to how pods are being rolled on deployments. Has anyone else encountered this, if so how was this resolved?

The solution may lie in how the kubernetes plugin registers/deregisters itself

**Comment by FarhanSajid1** [comment]
I'm guessing this is the solution, any recommendations on what to set this to?

https://github.com/asim/go-micro/blob/0d57213d3f5cbe75f3fd8de17f8de7715e1cfb2e/examples/heartbeat/README.md

**Comment by Davincible** [comment]
@FarhanSajid1 do you still have this issue?

The go-micro cli has implemented the gRPC heartbeat functionality now to support k8s probes.

**Comment by Scukerman** [comment]
I do and don't know how to fix it yet.

Every time I do the rollouts I encounter the timeouts. I believe the problem is in the default selector and registry. Looks like watcher doesn't work as expected. I need some time to figure out how to debug this.

**Comment by Davincible** [comment]
@Scukerman which registry are you using? I have no issue deploying in k8s with the k8s registry, and implemented the gRPC health protocol, which can be done by default with the `--health` flag in the CLI

**Comment by Scukerman** [comment]
@Davincible I just switched from micro 0.x to micro v3 and I still use old etcdv2 registry (ex etcd). I ported it to micro v3 and it seemed to work until I started rolling out the pods.

For now, every time I scale down or roll out the pods I get "Request Timeout" error for, like, 40 seconds and then it starts to work as expected.

I'll try to switch to etcd v3 version of the registry, and if this won't fix the problem, then I'll try k8s registry.

Anyway, I don't want to overload k8s's apiserver and etcd and I'm trying to avoid it.
